README.md

```diff
@@ -10,19 +10,95 @@ Assuming it has been published to RubyGems, you can install this gem using:
 you@machine $ inspec plugin install inspec-vault
 ```
 
+## Loading Secrets into Vault
+
+A full introduction to Vault is beyond the scope of this document, but begin by can downloading a recent version from https://www.vaultproject.io . Then, start a Vault dev-mode server with the following command:
+
+```
+$ vault server -dev
+```
+
+From there, you can then store an input. For example, look at the command below to store an input named `my_input` with the value of 2, for the profile `my_profile`. Once entered, Vault responds with metadata about the entry.
+
+```
+[cwolfe@lodi inspec-vault]$ vault kv put secret/inspec/my_profile my_input=2
+Key              Value
+---              -----
+created_time     2019-09-10T17:54:16.237055Z
+deletion_time    n/a
+destroyed        false
+version          1
+```
+
+With that value stored, Chef InSpec will now be able to retrieve the value.
+
 ## What This Plugin Does
 
-No idea.
+With the inspec-vault plugin enabled, Chef InSpec will contact the Vault server whenever an `input()` DSL call appears in profile control code. For example, whenever profile code like this is encountered:
+
+```ruby
+# In profile "my_profile"
+describe input("some_input") do
+  it { should cmp "some_expected_value" }
+end
+```
+
+Chef Inspec will determine a secret lookup path and access Vault. With no other settings, Chef InSpec will look for a Vault secret located at `secret/inspec/my_profile` with a key named "some_input". Chef InSpec will use the Vault secret if found, but otherwise it will fall back to other means of resolving the input, such as the profile metadata or CLI values.
```

Comment:
Signed-off-by: Mary Jinglewski <mjinglewski@chef.io>
```suggestion
Chef InSpec will determine a secret lookup path and access Vault. With no other settings, Chef InSpec will look for a Vault secret located at `secret/inspec/my_profile` with a key named `some_input`. Chef InSpec will use the Vault secret if found, but otherwise it will fall back to other means of resolving the input, such as the profile metadata or CLI values.
```